packages/nimble-components/src/table-column/base/tests/table-column.fixtures.ts

```diff
@@ -58,3 +60,59 @@ export class TableColumnDelegatesClickAndKeydown extends TableColumn {
         });
     }
 }
+
+/**
+ * Column validator used by TableColumnValidationTest
+ */
+export class TestColumnValidator extends ColumnValidator<
+readonly ['invalidFoo', 'invalidBar']
+> {
+    public constructor(columnInternals: ColumnInternals<unknown>) {
+        super(columnInternals, ['invalidFoo', 'invalidBar'] as const);
+    }
+
+    public validateFoo(isValid: boolean): boolean {
+        return this.setConditionValue('invalidFoo', !isValid);
+    }
+
+    public validateBar(isValid: boolean): boolean {
+        return this.setConditionValue('invalidBar', !isValid);
+    }
+}
+
+export const tableColumnValidationTestTag = 'nimble-test-table-column-validation';
+/**
+ * Test column type used to verify column config validation
+ */
+@customElement({
+    name: tableColumnValidationTestTag
+})
+export class TableColumnValidationTest extends TableColumn {
+    @attr()
+    public foo: boolean;
+
+    @attr()
+    public bar: boolean;
+
+    private readonly validator: TestColumnValidator;
+
+    public constructor(foo = true, bar = true) {
+        super({
+            cellRecordFieldNames: [],
+            cellViewTag: tableColumnEmptyCellViewTag,
+            groupHeaderViewTag: tableColumnEmptyGroupHeaderViewTag,
+            delegatedEvents: []
+        });
+        this.validator = new TestColumnValidator(this.columnInternals);
+        this.foo = foo;
+        this.bar = bar;
+    }
+
+    private fooChanged(): void {
+        this.validator.validateFoo(!!this.foo);
```

Comment:
Why is the double negation of `this.foo` needed here? If `this.foo` is always a `boolean` as defined on the class, I wouldn't think it is needed.

Same for `!!this.bar` below.

Reply:
I think it stems from a couple of unexpected patterns being used:
- Boolean attributes must be initialized with false.
- Custom element constructors should not take parameters.
- You should avoid instantiating custom elements by the constructor. They should be created by the browser via document.createElement or from an html template evaluation. (It's not strictly required from a technology perspective but it helps to avoid falling into anti-patterns like requiring the constructor to be invoked with specific parameters)

Reply:
Ok, no longer calling constructor (which doesn't take any args anymore).